File: src/main/java/model/Competencia.java
package model;

import org.hibernate.annotations.GenericGenerator;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Objects;

@Entity
public class Competencia implements Serializable {
    private static final long serialVersionUID = -2449149862024727141L;

    @Id
    @GeneratedValue(
            strategy = GenerationType.AUTO,
            generator = "native")
    @GenericGenerator(
            name = "native",
            strategy = "native")
    private Long id;

    @Column
    private String descricao;

    public Competencia() {
    }

    public Competencia(Competencia competencia) {
        this.id = competencia.id;
        this.descricao = competencia.descricao;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getDescricao() {
        return descricao;
    }

    public void setDescricao(String descricao) {
        this.descricao = descricao;
    }


    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Competencia that = (Competencia) o;
        return Objects.equals(id, that.id) &&
                Objects.equals(descricao, that.descricao);
    }

    @Override
    public int hashCode() {

        return Objects.hash(id, descricao);
    }

	@Override
	public String toString() {
		return descricao;
	}
}

File: src/main/java/model/Competencia.kt
package model

import org.hibernate.annotations.GenericGenerator
import java.io.Serializable
import javax.persistence.*

@Entity
class Competencia : Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO, generator = "native")
    @GenericGenerator(name = "native", strategy = "native")
    var id: Long = 0L
    @Column
    var descricao: String = ""

    constructor()

    constructor(competencia: Competencia) {
        this.id = competencia.id
        this.descricao = competencia.descricao
    }

}